src/app/model/services/projects.service.ts

```diff
@@ -12,62 +12,155 @@
+import {ProjectItem} from "../webapi/knora/v1/projects/project-item";
+import {ReplaySubject} from "rxjs/ReplaySubject";
+import {ProjectsList} from "../webapi/knora/v1/projects/projects-list";
+import {ProjectMembers} from "../webapi/knora/v1/projects/project-members";
+import {UserData} from "../webapi/knora/v1/users/user-data";
 
 
 @Injectable()
 export class ProjectsService extends ApiService {
 
+    private projects: ReplaySubject<ProjectItem[]> = new ReplaySubject(1);
+    private activeProject: ReplaySubject<ProjectItem> = new ReplaySubject(1);
+    private projectMembers: ReplaySubject<UserData[]> = new ReplaySubject(1);
+
     /**
-     * returns a project object
+     * returns a list of all projects
      *
-     * @param iri (= project iri)
-     * @returns {Observable<any>}
+     * @returns {ReplaySubject<ProjectItem[]>}
      */
-    getProjectByIri(iri: string): Observable<any> {
-        return this.httpGet("/projects/" + encodeURIComponent(iri));
+    getAllProjects(): ReplaySubject<ProjectItem[]> {
+
+        console.log("ProjectsService - getAllProjects");
+
+        this.httpGet("/projects")
+            .subscribe(
+                (result: ApiServiceResult) => {
+                    const projectResponse: ProjectsList = result.getBody(ProjectsList);
+                    const projects: ProjectItem[] = projectResponse.projects;
+                    // console.log("ProjectsService - getAllProjects - projects: " + JSON.stringify(projects));
+                    this.projects.next(projects);
+                },
+                (error: ApiServiceError) => {
+                    const errorMessage = <any>error;
+                    console.log("ProjectsService - getAllProjects - error: " + errorMessage);
+                    throw error;
+                }
+            );
+
+
+        return this.projects;
     }
 
     /**
      * returns a project object
      *
-     * @param shortname
-     * @returns {Observable<any>}
+     * @param iri (= project iri)
+     * @returns {ReplaySubject<ProjectItem>}
      */
-    getProjectByShortname(shortname: string): Observable<any> {
-        return this.httpGet("/projects/" + shortname + "?identifier=shortname");
+    getProjectByIri(iri: string): ReplaySubject<ProjectItem> {
+        console.log("ProjectsService - getProjectByIri: " + iri);
+
+        this.httpGet("/projects/" + encodeURIComponent(iri))
+            .subscribe(
+                (result: ApiServiceResult) => {
+                    const projectResponse: Project = result.getBody(Project);
+                    const project: ProjectItem = projectResponse.project_info;
+                    this.activeProject.next(project);
+                },
+                (error: ApiServiceError) => {
+                    const errorMessage = <any>error;
+                    console.log(errorMessage);
+                    throw error;
+                }
+            );
+
+        return this.activeProject;
```

Comment:
Here is the spot, I mentioned last week, where you could avoid doubling the code of the http request for project. Also it could be worth to consider using variables declared at the top of the class for the request path instead of strings repeated in every function (maybe in other services too, @kilchenmann ?). Because I could not test this, please feel free to omit or add some returns where necessary, also to name variables like you prefer.

```
getProjectByIri(iri: string): ReplaySubject<ProjectItem> {
    const path = "/projects/" + encodeURIComponent(iri); 
    // or better: const path = projectPath + encodeURIComponent(iri); 
    // with declaration of projectPath at the top of the class
    return this.getProject(path);
}

getProjectByShortname(shortname: string): ReplaySubject<ProjectItem> {
    const path = "/projects/" + shortname + "?identifier=shortname";
    // or better: const path = projectPath + shortname + shortIdentifier; 
    // with declaration of projectPath & shortIdentifier at the top of the class
    return this.getProject(path);
}

getProject(path: string): ReplaySubject<ProjectItem> {     // maybe delcared as private
    this.httpGet(path)
            .subscribe(
                (result: ApiServiceResult) => {
                    const projectResponse: Project = result.getBody(Project);
                    const project: ProjectItem = projectResponse.project_info;
                    this.activeProject.next(project);
                },
                (error: ApiServiceError) => {
                    const errorMessage = <any>error;
                    console.log(ProjectsService - getProject - error: " + errorMessage);
                    throw error;
                }
            );

        return this.activeProject;
}
```